### DISCUSSION TITLE
Linking material property to ADDirichletBC

### DISCUSSION THREAD

**Discussion by LukeFisherUKAEA** [idaholab/moose#23634] [Q&A Modules: General] [answered]
Hello,

I am looking to bring the output of a MaterialProperty block into an AD Dirichlet boundary condition and wondered if code for this already exists in the BCs library?  I can see that an ADMatNeumann BC exists, but cannot see the same for a Dirichlet condition.  Is there something fundamental (e.g. integrated vs nodal boundary condition type) that would prevent this from working?  For context, we are looking to implement Sieverts Law on a boundary:

![image](https://user-images.githubusercontent.com/91187320/223387645-15bb2244-beaa-44ad-bf9d-c5337f8e1dea.png)

where

![image](https://user-images.githubusercontent.com/91187320/223387697-f7449271-829f-49c6-aa1c-b48fe151f38a.png)

Initially temperature, T, will be implemented as an AuxVariable but we will want to get to the point where we can solve for heat transfer in parallel, meaning T will become a non-linear variable.  The intention would be to calculate the solubility coefficient in a ParsedMaterial block and feed that into a Dirichlet boundary condition to calculate surface concentration, C, with pressure also an AuxVariable.

Any advice or suggestions would be much appreciated.  Thank you.

**Comment by GiudGiud** [comment]
Hello

We do not have such a BC currently.
You may consider the ADMatchedValueBC for an example.

Using AD will simplify the implementation

Guillaume

**Comment by hugary1995** [comment, marked as answer]
> Is there something fundamental (e.g. integrated vs nodal boundary condition type) that would prevent this from working?

Correct. Material properties are stored on quadrature points, and nodal BCs are nodal... The two natural workarounds are:
1. convert the nodal BC to integrated BC, for example MatchedValueBC as @GiudGiud suggested.
2. project (extrapolate) the material properties onto the boundary nodes.

**Reply by GiudGiud** [reply]
Let's link those threads
https://github.com/idaholab/moose/discussions/24106

**Reply by lindsayad** [reply]
`ADMatchedValueBC` is also a nodal BC
